### ISSUE TITLE
Alternative way to receive notifications without Google services

### ISSUE THREAD

**Issue by kaladivo** [vexl-it/vexl#239]
Open a notofication

**Comment by leomira** [comment]
see the discussion at #235 #310

**Comment by kaladivo** [comment]
Blocked by #2094

**Comment by trymeouteh** [comment]
Would like this for degoogled users and to get this on F-Droid

**Comment by leomira** [comment]
@trymeouteh F-droid is out of our focus, even when we finish this we will not be pushing towards F-droid. 

Vexl is currently available via Obtainium, APK, Aurora, or even Zapstore.

**Comment by jarvisonsmart** [comment]
> 🤖 [Fable] on Dave's behalf.

## Plan: Vexl-owned background notification channel for Android without Google services

### Why not UnifiedPush (for now)

We evaluated [UnifiedPush](https://unifiedpush.org) (ntfy & co.) first. It is the right mechanism technically (battery-friendly, encrypted, what Element/Molly/SimpleX converged on), but it requires a distributor app on the device. Outside /e/OS (bundled ntfy fork) and people already running another UP app, that means "install ntfy first" — unacceptable friction for the users actually complaining here (de-googled phones/profiles, Huawei-type devices). We may add UP later as an optional, battery-friendlier channel; it is not part of this work.

### What we build

An Android-only, Vexl-owned background channel: a small native foreground service holds a WebSocket to the **existing** notification-service stream (`/rpc` `listenToNotifications`). No new server transport, no third-party app, no Google. iOS and the FCM path stay untouched.

Delivery chain on the server becomes: **foreground socket → background socket → push (FCM, if the device has it)**. For users without Google services the chain simply ends at the background socket.

### Server (notification-service) — minimal

- Additive `connectionKind: 'foreground' | 'background'` on the stream client info (absent = foreground, so old clients are unaffected).
- Stream-only chat messages (typing indicators) are not delivered to background connections.
- Everything else unchanged: multiple connections per secret already fan out, no throttling on sockets, existing 30 s "not delivered to any socket → push" timeout stays as is.
- `channel` attribute (`foreground_socket` / `background_socket` / `push`) on notification metrics; update `docs/backend_stats.md`; add `docs/notifications.md` documenting all channels.

### Mobile — new package `packages/expo-background-notification-socket`

Expo module (Kotlin) + config plugin; iOS is a no-op stub.

- Foreground service of type `remoteMessaging` (same as Signal's websocket mode), `START_STICKY`, restarted on `BOOT_COMPLETED` (to be verified against Android 15 FGS-from-boot rules; fallback is starting it on the first WorkManager tick).
- Kotlin WebSocket client speaking the `@effect/rpc` 0.72 NDJSON frames we already pin on both ends (`Request` → `Chunk`/`Ack` → `Exit`, `Ping`/`Pong`), reconnect with backoff, network-change aware.
- Secret, API URL, platform and app version stored in `EncryptedSharedPreferences` (written by JS, cleared on logout/disable) so the service can connect at boot before JS exists.
- Each received message starts a HeadlessJS task that runs the existing `processNotification` pipeline (new source `'backgroundSocket'`) — same handlers as FCM data pushes today (chat fetch, offer re-encryption, clubs, …). Protocol `Ack` is sent on receipt; processing outcome is reported via the existing `/report-notification`.
- When the app has a resumed activity, the native side skips the headless task — the foreground JS socket owns that case (both sockets may be connected at once; the server sends to both).
- Battery-optimisation exemption requested Signal-style (`REQUEST_IGNORE_BATTERY_OPTIMIZATIONS`), only when the channel is enabled.
- JS API: `configure`, `setEnabled`, `isEnabled`, `requestBatteryExemption`, `isGmsAvailable`.

### UX

- **No Google services detected**: enabled by default. One explanatory sheet at the existing notification-permission step (persistent notification + battery dialog explained), then the system battery-exemption dialog; `POST_NOTIFICATIONS` requested if missing.
- **Google services present**: off by default; toggle in notification settings (Android only). Expo token stays registered; the server chain makes push the automatic fallback.
- Persistent low-importance notification: "Vexl running on background".
- Settings: enable/disable toggle + "fix battery settings" row when the exemption is missing. Debug screen shows socket state.
- Existing hourly background sweep stays as last resort (also covers users who turn the channel off).

### Out of scope

iOS, FCM changes, UnifiedPush, F-Droid, separate build flavors (single APK for Play/APK/Obtainium/Zapstore).
